app/src/main/java/io/appium/settings/RecorderThread.java

```diff
@@ -0,0 +1,419 @@
+                    && muxerStarted && this.bufferInfo.size != 0 &&
+                    (this.bufferInfo.flags & MediaCodec.BUFFER_FLAG_CODEC_CONFIG) == 0) {
+                this.lastAudioTimestampUs = this.bufferInfo.presentationTimeUs;
+                muxer.writeSampleData(audioTrackIndex, encodedData, this.bufferInfo);
+            }
+
+            audioEncoder.releaseOutputBuffer(encoderStatus, false);
+
+            if ((this.bufferInfo.flags & MediaCodec.BUFFER_FLAG_END_OF_STREAM) != 0) {
+                Log.v(TAG, "buffer eos");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private boolean writeVideoBufferToFile() {
+        int encoderStatus;
+
+        encoderStatus = videoEncoder.dequeueOutputBuffer(bufferInfo,
+                RecorderConstant.MEDIA_QUEUE_BUFFERING_DEFAULT_TIMEOUT);
+        if (encoderStatus == MediaCodec.INFO_OUTPUT_FORMAT_CHANGED) {
+            if (videoTrackIndex > 0) {
+                Log.e(TAG, "videoTrackIndex less than zero");
+                return false;
+            }
+            videoTrackIndex = muxer.addTrack(videoEncoder.getOutputFormat());
+            startMuxerIfSetUp();
+        } else if (encoderStatus < 0 && encoderStatus != MediaCodec.INFO_TRY_AGAIN_LATER) {
+            Log.w(TAG, "unexpected result from encoder.dequeueOutputBuffer: "
+                    + encoderStatus);
+        } else if (encoderStatus >= 0) {
+            ByteBuffer encodedData = videoEncoder.getOutputBuffer(encoderStatus);
+            if (encodedData == null) {
+                Log.w(TAG, "videoEncoder, encodedData null");
+                return false;
+            }
+
+            if (bufferInfo.size != 0 &&
+                    (bufferInfo.flags & MediaCodec.BUFFER_FLAG_CODEC_CONFIG) == 0) {
+                bufferInfo.presentationTimeUs = getPresentationTimeUs();
+                muxer.writeSampleData(videoTrackIndex, encodedData, bufferInfo);
+            }
+
+            videoEncoder.releaseOutputBuffer(encoderStatus, false);
+
+            if ((bufferInfo.flags & MediaCodec.BUFFER_FLAG_END_OF_STREAM) != 0) {
+                Log.v(TAG, "videoEncoder, buffer eos");
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.Q)
+    @Override
+    public void run() {
+        try {
+            setupVideoCodec();
+            setupVirtualDisplay();
+
+            setupAudioCodec();
+            setupAudioRecord();
+
+            muxer = new MediaMuxer(this.outputFilePath,
+                    MediaMuxer.OutputFormat.MUXER_OUTPUT_MPEG_4);
+
+            // set output file orientation info if it is in landscape mode (default is 0==portrait)
+            // note: this method must be run before muxer.start()
+            if (isRotated) {
+                muxer.setOrientationHint(90);
```

Comment:
cannot we just read the number of rotation degrees from the system? I know there are some devices where 90 corresponds to portrait rather than landscape

Reply:
done 👍 i changed to dynamic approach, it cost us calling another system service https://github.com/sirmordred/io.appium.settings/blob/master/app/src/main/java/io/appium/settings/Settings.java#L132 but dynamic over hardcoded is the real solution yeah 👍